controllers/module_nmc_reconciler.go

```diff
@@ -135,6 +166,31 @@ func (mnrh *moduleNMCReconcilerHelper) getRequestedModule(ctx context.Context, n
 	return &mod, nil
 }
 
+func (mnrh *moduleNMCReconcilerHelper) finalizeModule(ctx context.Context, mod *kmmv1beta1.Module) error {
+	nmcList := kmmv1beta1.NodeModulesConfigList{}
+	err := mnrh.client.List(ctx, &nmcList)
+	if err != nil {
+		return fmt.Errorf("failed to list NMCs in the cluster: %v", err)
+	}
+
+	errs := make([]error, 0, len(nmcList.Items))
+	for _, nmc := range nmcList.Items {
+		err = mnrh.removeModuleFromNMC(ctx, &nmc, mod.Namespace, mod.Name)
+		errs = append(errs, err)
```

Comment:
```suggestion
		errs = append(
			errs,
			mnrh.removeModuleFromNMC(ctx, &nmc, mod.Namespace, mod.Name),
		)
```
Otherwise I think we keep appending the same `err` variable.

Reply:
No, just verified it is working without issues. The pointer are copied into the slice, so it is not an issue to replace the err value with a new pointer

Reply:
OK then.